libs/shared/ui/src/lib/components/inputs/input-select/input-select.tsx

```diff
@@ -1,125 +1,170 @@
-import { ListboxButton, ListboxInput, ListboxList, ListboxOption, ListboxPopover } from '@reach/listbox'
 import { useEffect, useState } from 'react'
+import Select, {
+  GroupBase,
+  MultiValue,
+  MultiValueProps,
+  OptionProps,
+  SingleValue,
+  SingleValueProps,
+  components,
+} from 'react-select'
 import { Value } from '@console/shared/interfaces'
 import Icon from '../../icon/icon'
-import InputSearch from '../input-search/input-search'
 
 export interface InputSelectProps {
-  label: string
-  value?: string | undefined
-  items: Value[]
   className?: string
-  onChange?: (e: string) => void
-  error?: string
+  label: string
+  value?: string | string[]
+  options: Value[]
   disabled?: boolean
-  portal?: boolean
+  error?: string
+  onChange?: (e: string | string[]) => void
   dataTestId?: string
-  search?: boolean
+  isMulti?: true
+  portal?: boolean
+  isSearchable?: boolean
+  isClearable?: boolean
 }
 
 export function InputSelect(props: InputSelectProps) {
   const {
+    className = '',
     label,
     value,
-    items,
-    className = '',
+    options,
+    disabled,
+    error = false,
     onChange,
-    error,
     dataTestId,
-    disabled = false,
-    portal = true,
-    search,
+    isMulti = undefined,
+    isSearchable = false,
+    isClearable = false,
   } = props
+  const [focused, setFocused] = useState(false)
+  const [selectedItems, setSelectedItems] = useState<MultiValue<Value> | SingleValue<Value>>([])
+  const [selectedValue, setSelectedValue] = useState<string | string[]>([])
 
-  const [selectedValue, setSelectedValue] = useState(value && items.find((item) => item.value === value))
-  const hasError = error && error.length > 0 ? 'input--error' : ''
+  const hasFocus = focused
+  const hasError = error ? 'input--select-multiple--error' : ''
 
-  const [currentItems, setCurrentItems] = useState(items)
+  const handleChange = (values: MultiValue<Value> | SingleValue<Value>) => {
+    setSelectedItems(values)
 
-  useEffect(() => {
-    if (value) setSelectedValue(items.find((item) => item.value === value))
-  }, [value, items, setSelectedValue])
+    if (isMulti) {
+      const vals = (values as MultiValue<Value>).map((value) => value.value)
+      onChange && onChange(vals)
+      setSelectedValue(vals)
+    } else {
+      if (values) {
+        onChange && onChange((values as Value).value)
+        setSelectedValue((values as Value).value)
+      }
+    }
+  }
 
   useEffect(() => {
-    setCurrentItems(items)
-  }, [items])
+    const items = options.filter((option) => {
+      if (isMulti) {
+        return (value as string[])?.includes(option.value)
+      } else {
+        if (Object.prototype.toString.call(value) === '[object Array]') {
```

Comment:
Does it work like `type of` ? (maybe more simple to use type of)

Reply:
I changed fo `Array.isArray(value)` https://stackoverflow.com/questions/4775722/how-can-i-check-if-an-object-is-an-array